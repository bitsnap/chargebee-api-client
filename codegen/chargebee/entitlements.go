package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListEntitlements generates chargebee client code to fetch all items
//
// API: https://{site}.chargebee.com/api/v2/entitlements
//
// Documentation: https://apidocs.chargebee.com/docs/api/entitlements?lang=curl#list_all_entitlements
func GenerateListEntitlements() string {
	return client.GenerateList("Entitlements", "Entitlement", "chargebee.com/api/v2/entitlements")
}

// GenerateEntitlementsModel generates chargebee entitlement domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/entitlements?lang=curl#entitlement_attributes
func GenerateEntitlementsModel() string {
	return client.GenerateModel("Entitlement", "Entitlement attributes", "https://apidocs.chargebee.com/docs/api/entitlements")
}
