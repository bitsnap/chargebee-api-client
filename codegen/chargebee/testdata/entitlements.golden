# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee entitlements code generation when generating Chargebee API client should be able to generate entitlements models" = '''
(string) (len=317) "type Entitlement struct {\n    Id string `json:\"id\" validate:\"required\"`\n    EntityId string `json:\"entity_id\"`\n    EntityType enums.EntityTypeEnum `json:\"entity_type\"`\n    FeatureId string `json:\"feature_id\"`\n    FeatureName string `json:\"feature_name\"`\n    Value string `json:\"value\"`\n    Name string `json:\"name\"`\n}"
'''
